app/helpers/container_summary_helper.rb

```diff
@@ -1,142 +1,126 @@
-      "container_service",
-      "container_service_show_list"
-    )
+    textual_link(@record.container_services)
   end
 
   def textual_container_replicators
-    textual_multiple_relationship(
-      :container_replicators,
-      "container_replicator",
-      "container_replicator_show_list"
-    )
+    textual_link(@record.container_replicators)
   end
 
   def textual_container_groups
-    textual_multiple_relationship(
-      :container_groups,
-      "container_group",
-      "container_group_show_list"
-    )
+    textual_link(@record.container_groups)
   end
 
   def textual_containers
-    textual_multiple_relationship(
-      :containers,
-      "container",
-      "containers",  # should it be container_show_list?
-    )
+    textual_link(@record.containers, :feature => "containers") # should it be container_show_list?
   end
 
   def textual_container_nodes
-    textual_multiple_relationship(
-      :container_nodes,
-      "container_node",
-      "container_node_show_list",
-    )
+    textual_link(@record.container_nodes)
   end
 
   def textual_container_node
-    textual_single_relationship(
-      :container_node,
-      :name,
-      "container_node",
-      "container_node_show",
-    )
+    textual_link(@record.container_node)
   end
 
   private
 
-  def textual_single_relationship(entity, attribute, image, feature, controller = nil)
-    controller ||= entity.to_s
+  def textual_link(target, **opts, &blk)
+    case target
+    when ActiveRecord::Relation
+      textual_collection_link(target, **opts, &blk)
+    else
+      textual_object_link(target, **opts, &blk)
+    end
+  end
+
+  def textual_object_link(object, as: nil, feature: nil)
+    return if object.nil?
+
+    klass = as || object.class.base_model
 
-    label = ui_lookup(:table => controller)
-    rel_entity = @record.send(entity)
+    feature ||= "#{klass.name.underscore}_show"
 
-    return if rel_entity.nil?
-    rel_value = rel_entity.send(attribute)
+    label = ui_lookup(:class => klass.name)
+    image = textual_object_icon(object)
+    value = if block_given?
+              yield object
+            else
+              object.name
+            end
 
-    h = {:label => label, :image => image, :value => rel_value}
+    h = {:label => label, :image => image, :value => value}
 
     if role_allows(:feature => feature)
-      h[:link] = url_for(:controller => controller, :action => 'show', :id => rel_entity)
-      h[:title] = "Show #{label} '#{rel_value}'"
+      h[:link] = url_for(:controller => klass.name.underscore,
+                         :action     => 'show',
+                         :id         => object)
+      h[:title] = "Show #{label} '#{value}'"
     end
 
     h
   end
 
-  def textual_multiple_relationship(entity, image, feature)
-    label = ui_lookup(:tables => entity.to_s)
-    rel_num = @record.number_of(entity)
+  def textual_collection_link(collection, as: nil, feature: nil)
+    klass = as || collection.klass.base_model
```

Comment:
Nope, the method on a AR collection object to get the class that it's referring to is named "klass"